spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/SetStatusGatewayFilterFactory.java

```diff
@@ -62,11 +71,24 @@ public GatewayFilter apply(Config config) {
 			return chain.filter(exchange).then(Mono.fromRunnable(() -> {
 				// check not really needed, since it is guarded in setStatusCode,
 				// but it's a good example
-				setResponseStatus(exchange, statusHolder);
+				boolean isStatusCodeUpdated = setResponseStatus(exchange, statusHolder);
+				if (isStatusCodeUpdated && originalStatusHeaderName != null) {
+					HttpStatus statusCode = exchange.getResponse().getStatusCode();
```

Comment:
Don't you need to get the response status code **before** it is set above? Also, this will have problems with custom status codes (ie any not understood by `HttpStatus`).

Reply:
sorry,
Yesterday I saw this mistake and decided that it's time to sleep, to stop doing mistakes(
I've just fixed it anyway